### PR TITLE
Bulk Anticipation

### PR DESCRIPTION
**Pull request by thufik**
Antecipações implementadas

### CODE REVIEW

File: PagarMe.Shared/Model/Recipient.cs [hunk 1]
```diff
@@ -25,12 +25,15 @@
...
 using Newtonsoft.Json;
+using PagarMe.Model;
+using PagarMe.Base;
 
 namespace PagarMe
 {
 	public class Recipient : Base.Model
 	{
-		protected override string Endpoint { get { return "/recipients"; } }
+        private Base.ModelCollection<BulkAnticipation> _anticipations;
```
Comment (rodrigoPgSup) [inline PagarMe.Shared/Model/Recipient.cs]: Indentation

File: PagarMe.Shared/Model/Recipient.cs [hunk 2]
```diff
@@ -74,8 +77,79 @@ public double AnticipatableVolumePercentage
 			set { SetAttribute("anticipatable_volume_percentage", value); }
 		}
 
+        private BulkAnticipationLimit AnticipationLimits(TimeFrame timeframe, DateTime paymentDate)
```
Comment (rodrigoPgSup) [inline PagarMe.Shared/Model/Recipient.cs]: Indentation for All!!

File: PagarMe.Shared/Model/Recipient.cs [hunk 3]
```diff
@@ -74,8 +77,79 @@ public double AnticipatableVolumePercentage
...
+        {
+            BulkAnticipationLimit bulk = new BulkAnticipationLimit();
+            bulk.TimeFrame = TimeFrame.Start;
```
Comment (rodrigoPgSup) [inline PagarMe.Shared/Model/Recipient.cs]: Always Timeframe = Start ?

File: PagarMe.Shared/PagarMeRequest.cs [hunk 4]
```diff
@@ -76,6 +76,7 @@ public PagarMeResponse Execute()
         {
             HttpWebResponse response;
             var request = GetRequest();
+            //request.Method = Method;
```
Comment (rodrigoPgSup) [inline PagarMe.Shared/PagarMeRequest.cs]: Should be commented?

File: PagarMe.Tests/RecipientTests.cs [hunk 5]
```diff
@@ -31,8 +32,121 @@ public void CreateWithOldFields ()
...
+        public void ReturnAnticipationMaxValue()
+        {
+            Recipient recipient = CreateRecipient();
+            recipient.Save();
+
+            DateTime date = DateTime.Now;
+            date = date.AddDays(5);
+
+            var limit = recipient.MaxAnticipationValue(TimeFrame.Start, date);
+            Assert.IsNotNull(limit);
```
Comment (rodrigoPgSup) [inline PagarMe.Tests/RecipientTests.cs]: Maybe compare the MaxValue with Zero?

File: PagarMe.Tests/RecipientTests.cs [hunk 6]
```diff
@@ -31,8 +32,121 @@ public void CreateWithOldFields ()
...
+        public void ReturnAnticipationMinValue()
+        {
+            Recipient recipient = CreateRecipient();
+            recipient.Save();
+
+            DateTime date = DateTime.Now;
+            date = date.AddDays(5);
+
+            var limit = recipient.MinAnticipationValue(TimeFrame.Start, date);
+            Assert.IsNotNull(limit);
```
Comment (rodrigoPgSup) [inline PagarMe.Tests/RecipientTests.cs]: Idem MAX

File: Playground/Program.cs [hunk 7]
```diff
@@ -83,25 +83,25 @@ public static void Main(string[] args)
...
-                        Liable = false
-                    },
-                    new SplitRule {
-                        Recipient = r3,
-                        Percentage = 50,
-                        ChargeProcessingFee = false,
-                        Liable = false
-                    }
-                };
+                                new SplitRule {
```
Comment (rodrigoPgSup) [inline Playground/Program.cs]: Indent!

File: Playground/Program.cs [hunk 8]
```diff
@@ -112,6 +112,7 @@ public static void Main(string[] args)
                 foreach (var erro in ex.Error.Errors)
                     Console.WriteLine(String.Format("Error: {0}", erro.Message));
             }
-        }        
+
+        }
     }
 }
```
Comment (rodrigoPgSup) [inline Playground/Program.cs]: Breakline!

File: PagarMe.Pcl.nuspec [hunk 9]
```diff
@@ -2,7 +2,7 @@
 <package>
 	<metadata>
 	<id>Portable.Pagar.me</id>
-	<version>2.3.7</version>
+	<version>2.3.8</version>
```
Comment (rodrigoPgSup) [inline PagarMe.Pcl.nuspec]: Don't forget to do the same for PagarMe.nuspec

File: PagarMe.Shared/Utils.cs [hunk 10]
```diff
@@ -56,6 +56,13 @@ public class Utils {
 			}
 			return false;
 		}
+
```
Comment (henriquekano) [inline PagarMe.Shared/Utils.cs]: That indentation (for all)

File: PagarMe.Shared/PagarMeRequest.cs [hunk 11]
```diff
@@ -177,7 +177,6 @@ private HttpWebRequest GetRequest()
         {
             HttpWebRequest request = WebRequest.CreateHttp(GetRequestUri());
```
Comment (henriquekano) [inline PagarMe.Shared/PagarMeRequest.cs]: Please don´t include unnecessary changes

File: PagarMe.Shared/Model/BulkAnticipation.cs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        public int AnticipationFee
+        {
+            get { return GetAttribute<int>("anticipation_fee"); }
+        }
+
+        public bool Build
+        {
+            set { SetAttribute("build", value); }
```
Comment (henriquekano) [inline PagarMe.Shared/Model/BulkAnticipation.cs]: What would happen if the user sets 'build' to false? Would the created BulkAnticipation have the 'Building' status?

File: PagarMe.Shared/Model/BulkAnticipationLimit.cs [hunk 13]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PagarMe.Model
+{
+    internal class BulkAnticipationLimit : Base.Model
```
Comment (henriquekano) [inline PagarMe.Shared/Model/BulkAnticipationLimit.cs]: By extending the Base.Model wouldn´t the BulkAnticipationLimit have the save method?

File: PagarMe.Shared/Model/Limit.cs [hunk 14]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PagarMe.Model
+{
+    public class Limit : Base.Model
```
Comment (henriquekano) [inline PagarMe.Shared/Model/Limit.cs]: Same as the above comment

File: PagarMe.Shared/Model/Recipient.cs [hunk 15]
```diff
@@ -25,65 +25,139 @@
 // THE SOFTWARE.
 using System;
 using Newtonsoft.Json;
+using PagarMe.Model;
+using PagarMe.Base;
 
 namespace PagarMe
 {
-	public class Recipient : Base.Model
```
Comment (henriquekano) [inline PagarMe.Shared/Model/Recipient.cs]: Is it really necessary to correct all indentation in the file? It makes harder to view the significant changes to the feature

File: PagarMe.Shared/Model/BulkAnticipation.cs (hunk 12 above)
Comment (thufik) [inline PagarMe.Shared/Model/BulkAnticipation.cs]: No, of course it would have pending status as tested

File: PagarMe.Shared/Model/BulkAnticipationLimit.cs (hunk 13 above)
Comment (thufik) [inline PagarMe.Shared/Model/BulkAnticipationLimit.cs]: This class is internal, so it is not accessible to the user

File: PagarMe.Shared/Model/Limit.cs (hunk 14 above)
Comment (thufik) [inline PagarMe.Shared/Model/Limit.cs]: Good communication, I will make the necessary modifications.

File: PagarMe.Shared/Utils.cs (hunk 10 above)
Comment (thufik) [inline PagarMe.Shared/Utils.cs]: Ok

File: PagarMe.Shared/Model/Recipient.cs (hunk 15 above)
Comment (thufik) [inline PagarMe.Shared/Model/Recipient.cs]: Yes, because this particular file did not conform to the project's indentation, so the total change was necessary. Although there have been no changes outside this

File: PagarMe.Shared/Model/Limit.cs (hunk 14 above)
Comment (thufik) [inline PagarMe.Shared/Model/Limit.cs]: We are not dealing with the existence of the save method in any model class yet, this will be a restructuring that will be done in Lib later.
